velox/exec/fuzzer/WindowFuzzer.cpp

```diff
@@ -227,23 +382,75 @@ void WindowFuzzer::go() {
     const auto call =
         makeFunctionCall(signature.name, argNames, false, false, ignoreNulls);
 
-    std::vector<SortingKeyAndOrder> sortingKeysAndOrders;
-    // 50% chance without order-by clause.
-    if (vectorFuzzer_.coinToss(0.5)) {
-      sortingKeysAndOrders =
-          generateSortingKeysAndOrders("s", argNames, argTypes);
-    }
+    auto [frameType, startBoundType, endBoundType] =
+        frameWindowTypeAndBoundType();
+
+    auto isKBoundFrame = [&](core::WindowNode::BoundType boundType) {
+      return (
+          boundType == core::WindowNode::BoundType::kPreceding ||
+          boundType == core::WindowNode::BoundType::kFollowing);
+    };
+
+    bool isKRangeFrame = frameType == core::WindowNode::WindowType::kRange &&
+        (isKBoundFrame(startBoundType) || isKBoundFrame(endBoundType));
+    auto hasRowNumberKey =
+        requireSortedInput && frameType == core::WindowNode::WindowType::kRows;
+
     const auto partitionKeys = generateSortingKeys("p", argNames, argTypes);
-    const auto [frameClause, isRowsFrame] = generateFrameClause();
-    const auto input = generateInputDataWithRowNumber(
-        argNames, argTypes, partitionKeys, signature);
+    std::vector<SortingKeyAndOrder> sortingKeysAndOrders;
     // If the function is order-dependent or uses "rows" frame, sort all input
     // rows by row_number additionally.
-    if (requireSortedInput || isRowsFrame) {
+    if (hasRowNumberKey) {
       sortingKeysAndOrders.emplace_back("row_number", core::kAscNullsLast);
       ++stats_.numSortedInputs;
     }
 
+    TypeKind orderByTypeKind;
+    // kRange frames need only one order by key. This should be row_number for
+    // functions that are order dependent.
+    if (isKRangeFrame) {
+      sortingKeysAndOrders = generateSortingKeysAndOrders(
+          "s", argNames, argTypes, hasRowNumberKey, isKRangeFrame);
+      orderByTypeKind = argTypes.back()->kind();
+    } else if (vectorFuzzer_.coinToss(0.5)) {
+      // 50% chance without order-by clause.
+      sortingKeysAndOrders =
+          generateSortingKeysAndOrders("s", argNames, argTypes);
+    }
+
+    auto input = generateInputDataForWindowFuzzer(
+        argNames, argTypes, partitionKeys, signature, hasRowNumberKey);
+    if (isKRangeFrame) {
+      // Catch possible type overflow errors when generating offset columns.
+      try {
+        auto orderByKey = sortingKeysAndOrders.back();
```

Comment:
Since this should have exactly one key, you should add a VELOX_CHECK for the size of sortingKeysAndOrders and pick up the value at the first index instead of using back().